spyder/plugins/ipythonconsole.py

```diff
@@ -861,12 +861,19 @@ def create_new_client(self, give_focus=True):
         """Create a new client"""
         self.master_clients += 1
         name = "%d/A" % self.master_clients
+        cf = self._new_connection_file()
+        if cf is None:
+            QMessageBox.warning(self, _('Warning'),

```

Comment:
> however I can't use the client.show_kernel_error because in order to create the client the conecction file has to be not None.

That's not right. `ClientWidget` can be instantiated without a connection file and also without a kernel.

So please move the validation of `cf is None` just after `add_tab` is called and use `client.show_kernel_error` to show our new error message.